### PR TITLE
Component slots

### PR DESCRIPTION
**Pull request by sroussey**
Rebased to main and then added more slots and made them available to custom nodes.

### CODE REVIEW

**Comment by clarkmcc** [comment]
Is this ready for review? I didn't want to dive in if you're not happy with it yet, but I did see your reference comment on the other issue.

**Comment by sroussey** [comment]
Yeah, it is working so far. 

I have more ideas, but don't think I'll get to them for a while.

File: lib/node-builder.tsx [hunk 1]
```diff
@@ -21,6 +21,52 @@ import { NodeContainer } from './components/NodeContainer'
...
+  isFocused: boolean
+  onFocus: () => void
+  onBlur: () => void
+}
+export interface NodeBodySlots {
+  bodyTop?: React.ComponentType<NodeFocusState>
+  bodyBottom?: React.ComponentType<NodeFocusState>
+  inputs: React.JSX.Element[]
+  outputs: React.JSX.Element[]
+  inputGroups: React.JSX.Element[]
```
Comment (clarkmcc) [inline lib/node-builder.tsx]: What's the use-case for distinguishing between `inputs` and `inputGroups`? Can they be safely treated as the same thing?

**Review by clarkmcc — CHANGES_REQUESTED** [review]
Can you run prettier? There's a config already in the project.

File: lib/node-builder.tsx (hunk 1 above)
Comment (sroussey) [reply lib/node-builder.tsx]: oh, sure. I was really just replicated what you had. Not sure anyone would ever want to reverse them!

**Comment by sroussey** [comment]
> Can you run prettier? There's a config already in the project.

Really? I've been going nuts without it since there is no .vscode folder for this project

File: lib/node-builder.tsx (hunk 1 above)
Comment (clarkmcc) [reply lib/node-builder.tsx]: I think they need to be generated separately, but it sounds like they can safely just be combined into a single inputs slot

**Comment by clarkmcc** [comment]
Yeah I don't use vscode but having the config in the project is fine with me. I do think there's a few IDE-agnostic config files all at the root including a prettier setup.

Thanks for running the formatter! Will take a final look at this tomorrow.

**Review by clarkmcc — APPROVED** [review]

**Pull request merged**